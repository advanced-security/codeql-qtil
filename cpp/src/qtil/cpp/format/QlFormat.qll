private import qtil.format.QLFormat
private import cpp
private import qtil.cpp.locations.Locatable
import QlFormat<Location, CppLocatableConfig>
