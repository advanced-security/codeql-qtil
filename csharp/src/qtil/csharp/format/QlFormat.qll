private import qtil.format.QLFormat
private import csharp
private import qtil.csharp.locations.Locatable
import QlFormat<Location, CsharpLocatableConfig>
