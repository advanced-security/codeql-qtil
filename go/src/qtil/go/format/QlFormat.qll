private import qtil.format.QLFormat
private import go
private import qtil.go.locations.Locatable

import QlFormat<DbLocation, GoLocatableConfig>