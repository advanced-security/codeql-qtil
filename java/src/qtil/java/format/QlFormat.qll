private import qtil.format.QLFormat
private import java
private import qtil.java.locations.Locatable

import QlFormat<Location, JavaLocatableConfig>