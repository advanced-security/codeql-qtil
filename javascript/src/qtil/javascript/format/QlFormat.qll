private import qtil.format.QLFormat
private import javascript
private import qtil.javascript.locations.Locatable

import QlFormat<DbLocation, JavascriptLocatableConfig>