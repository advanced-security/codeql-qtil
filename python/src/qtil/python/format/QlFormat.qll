private import qtil.format.QLFormat
private import python
private import qtil.python.locations.Locatable
import QlFormat<Location, PythonLocatableConfig>
