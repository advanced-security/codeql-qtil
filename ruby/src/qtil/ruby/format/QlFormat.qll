private import qtil.format.QLFormat
private import ruby
private import qtil.ruby.locations.Locatable

import QlFormat<Location, RubyLocatableConfig>