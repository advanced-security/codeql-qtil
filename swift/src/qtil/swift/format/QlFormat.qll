private import qtil.format.QLFormat
private import swift
private import qtil.swift.locations.Locatable

import QlFormat<Location, SwiftLocatableConfig>