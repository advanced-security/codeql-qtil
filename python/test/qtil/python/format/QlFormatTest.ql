import python
import qtil.python.format.QlFormat

predicate problem(AstNode elem, Template msg) {
  exists(Assign assign, Expr value, Variable var |
    elem = assign and
    assign.defines(var) and
    value = assign.getValue() and
    msg =
      tpl("Variable {varname} defined with value {init}.")
          .text("varname", var.getId())
          .link("init", value)
  )
}

import Problem<problem/2>::Query