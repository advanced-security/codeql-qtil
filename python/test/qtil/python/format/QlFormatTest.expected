| test.py:1:1:1:5 | AssignStmt | Variable x defined with value $@. | test.py:1:5:1:5 | test.py:1:5:1:5 | IntegerLiteral | file://:0:0:0:0 | (none) | [unused] |
